File: scripts/count-cpu/count-cpu.pl
#!/usr/bin/env perl

use v5.38;
use System::CPU;

say "Using System::CPU, count: ", System::CPU::get_ncpu;

open (my $FH, '<', '/proc/cpuinfo') or die "Can't open /proc/cpuinfo: $!";
my $cores = 0;
while (<$FH>) { $cores++ if /^processor\s*:/ }
close $FH;
say "Reading /proc/cpuinfo, count: $cores";

File: scripts/count-cpu/count-cpu.py
import os

print("Using os, count: ", os.cpu_count())

import multiprocessing

print("Using multiprocessing, count: ", multiprocessing.cpu_count())

def cpu_count():
    try:
        with open('/proc/cpuinfo', 'r') as fh:
            cpuinfo = fh.readlines()
        cores = sum(1 for line in cpuinfo if line.startswith('processor'))
        return cores
    except FileNotFoundError:
        return None

print("Reading /proc/cpuinfo, count: ", cpu_count())
